### PR TITLE
SeasonTST finetuning

### PR DESCRIPTION
**Pull request by paololucchino**
This PR:

1) adds a some general enhancements to the training workflow for SeasonTST (and indeed any gridded dataset):
- Adds some code explanation and readmes to help unpack the internals of the model.
- Includes changes from #4 
- Adds logging in `learner` to debug and measure training times.
- Adds `drop_last` and `prefetch_factor` to dataloader and includes optimizations from #5 
- Add feature to choose custom value for masking for pretraining or imputaiton task.
- Adds standard scaling functionality to `TimeLatLonDataset`

2) Adds first version of a SeasonTST finetuning script for the _forecasting_ task
- Note the importance of the `exclude_head` argument in `transfer_weights` which should be set to False when starting finetuning from a previous finetuning checkpoint
- Ran this for ~60 epochs for 2 pixels and confirm the model is learning (training and validation loss fall)

3) Implements partial functionality for the _imputation_ task:
- It add `ObservationMaskCB` to randomly mask observations in the training phase, to get the model to learn the task of reconstructing missing or lagging values. 

Considerations for future development PRs: 
- With hindsight, randomly masking for the imputation task may more easily have been tackled within the Dataset rather than in a callback, and we might wish to switch to that in the future.
- Losses are currently printed after each epoch. With large datasets, this will mean infrequent insight into the training metrics. We should add logging of metrics after each training step.
- Next main development step is running the inference on the test data and reporting metrics.

### CODE REVIEW

**Review by sem0029 — APPROVED** [review]
Looks good to me, pretraining and finetuning both work on my machine! For the JH run I'll just create a new branch if things need changed for it. Thanks!

File: SeasonTST/utils.py [hunk 1]
```diff
@@ -17,25 +17,28 @@ def get_dls(
...
         datasetCls=dataset_class,
         dataset_kwargs={
             "dataset": dataset,
+            "mask": mask,
             "size": size,
             "scale": True,
         },
         batch_size=config_obj.batch_size,
         workers=config_obj.num_workers,
+        prefetch_factor=config_obj.prefetch_factor
```
[the remark this answers is not in the document]
Comment (paololucchino) [reply SeasonTST/utils.py]: Should work subject to the `prefetch_factor` being added like this https://github.com/WFP-VAM/PatchTST/pull/3/files#diff-fba98138cfcf11011002d1e7e78b8f8d00f3a25cb2f92f5ec0a9d4d5b9bef66dR98. Is this still giving you the error?
